src/lib/components/Search/index.js

```diff
@@ -300,42 +203,44 @@ class Search extends BaseStateElement {
     this.animationTime = parseInt(value, 10);
   }
 
-  onKeyDown(e) {
-    // Check if the user is navigating within the search popout.
-    switch (e.key) {
-      case 'Home':
-        e.preventDefault();
-        this.firstHit();
-        return;
-
-      case 'End':
-        e.preventDefault();
-        this.lastHit();
-        return;
-
-      case 'Up': // IE/Edge specific value
-      case 'ArrowUp':
-        e.preventDefault();
-        this.prevHit();
-        return;
-
-      case 'Down': // IE/Edge specific value
-      case 'ArrowDown':
-        e.preventDefault();
-        this.nextHit();
-        return;
-
-      case 'Enter':
-        const hit = this.hits[this.cursor];
-        if (hit) {
-          this.navigateToHit(hit);
-        }
+  /**
+   * Keep track of which result is selected in the search results element and
+   * reflect them to aria-activedescendant.
+   * This ensures screen readers properly announce the current search result.
+   * We do this because focus never leaves the search input box, so when the
+   * user is arrowing through results, we have to tell the screen reader about
+   * it.
+   * @param {Event} event Select event fired by search results element.
+   */
+  onResultSelect(event) {
+    if (event.target instanceof HTMLElement) {
```

Comment:
I think you can remove this line and instead do `event.currentTarget` which will always be the results element because that's the one you've added the listener to.

Reply:
This is actually here only to make typescript happy, otherwise it complains that EventTarget does not have HTMLElement interface.

Reply:
oh I think you can do a cast with a comment. It's something like...

if ((/** @type {HTMLElement} */ event).target)